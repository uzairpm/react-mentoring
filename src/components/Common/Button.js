import React, { Component } from 'react';
import classname from 'classname';

const CLASS_NAME = 'btn';

export default class Button extends Component {
    constructor(props) {
        super(props);
        this.state = {
            disabled: false
        };
    }
    onClick(event) {
        const { clickHandler } = this.props;
        event.preventDefault();
        event.stopPropagation();
        clickHandler();
    }
    render() {
        const {
            type,
            style,
            className,
            disabled,
            label
        } = this.props;
        const classNames = classname(className, CLASS_NAME);

        return (
            <button
                type={type}
                className={classNames}
                style={style}
                disabled={disabled === true ? disabled : this.state.disabled}
                onClick={event => this.onClick(event)}>
                { label }
            </button>
        );
    }
}
