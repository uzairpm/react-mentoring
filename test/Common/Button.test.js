import { shallow } from 'enzyme';
import React from 'react';

import Button from '../../src/components/Common/Button';

describe('Check Component creation', () => {
    const getComponent = (props) => shallow(<Button label="submit" {...props} />);
    const element = getComponent({ type: 'submit', className: 'b1' });
    
    test('Button component snapshot test', () => {
        expect(element).toMatchSnapshot();
    });
    test('Testing for Button props', () => {
        expect(element.prop('type')).toEqual('submit');
        expect(element.prop('className')).toEqual('b1 btn');
    });
    test('Test if button is getting disabled', () => {
        const element2 = getComponent({ className: 'btn-primary', disabled: true });
        expect(element2.prop('disabled')).toEqual(true);
    });
    test('Test if Button handler is invoked', () => {
        const onClick = jest.fn();
        const element2 = shallow(<Button label='Submit' className='btn-primary' disabled={false} clickHandler={onClick}/>);
        expect(element2.prop('disabled')).toEqual(false);
        
        element2.find('button').simulate('click', {
            target: { },
            preventDefault: function() { },
            stopPropagation: function() { }
        });
        expect(onClick).toHaveBeenCalled();
    });
});