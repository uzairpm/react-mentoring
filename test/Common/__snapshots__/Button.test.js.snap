// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Component creation Button component snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    className="b1"
    label="submit"
    type="submit"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "submit",
      "className": "b1 btn",
      "disabled": false,
      "onClick": [Function],
      "style": undefined,
      "type": "submit",
    },
    "ref": null,
    "rendered": "submit",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "submit",
        "className": "b1 btn",
        "disabled": false,
        "onClick": [Function],
        "style": undefined,
        "type": "submit",
      },
      "ref": null,
      "rendered": "submit",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
